### DISCUSSION TITLE
How to fit the font sizes on Android tv emulator?

### DISCUSSION THREAD

**Discussion by avishka-codimite** [react-native-tvos/react-native-tvos#343] [Q&A] [answered]
when running the application on an Android tv emulator its shows a large scale of fonts. but I try to scale down the font size on settings but the issue is the same. Is there a way to get the correct sizes of fonts on Android TV emulators like Apple tv? if anyone facing this issue can help me to figure it out.

![Screenshot 2022-09-07 at 10 44 25 AM](https://user-images.githubusercontent.com/111336101/188797977-48b2471a-e110-407a-a69d-2b6169f17273.png)

I changed the font scaling size through settings but it didn't work. The small size of scale didn't work.

![Screenshot_1662527729](https://user-images.githubusercontent.com/111336101/188798140-817bcbd0-3201-42bc-809f-d31e7f9e0c16.png)

**Comment by douglowder** [comment]
I will have a look.

**Comment by douglowder** [comment, marked as answer]
@avishka-codimite I tested with the test app (RNTester) that ships with the react-native repo. At least on the Android TV emulator, the text sizes do appear to scale with the system text size setting. If you have an app that shows a font scaling issue on Android TV, please provide that and I'll be happy to investigate.

**Reply by avishka-codimite** [reply]
@douglowder Thank your feedback. Now I was able to fix that issue. I implemented a simple function to change the scale of each element. When using this was able to fit the font sizes and components. I tested it in a real environment and seems to be it's working.

```
import {Platform} from 'react-native';

//should pass the tvos size
const scale = size => (Platform.OS === 'android' ? size / 2 : size);

export {scale};
```

```
import {scale} from './Scaling';

export const styles = StyleSheet.create({
 description: {
    fontFamily: FONTS.medium,
    fontSize: scale(28),
    color: '#fff',
  },
});
```
